openstudiocore/src/model/ThermalZone.cpp

```diff
@@ -998,14 +993,32 @@ namespace detail {
 
   bool ThermalZone_Impl::setThermostat(const Thermostat & thermostat)
   {
-    resetThermostat();
-    return setPointer(OS_ThermalZoneFields::ThermostatName, thermostat.handle());
+    auto t_model = model();
+
+    if( t_model != thermostat.model() ) {
```

Comment:
Yeah and the other thing is that the set method return value can come too late. Like say for instance if you do a reset first.